packages/xcoobee-cookie-kit-react/src/BridgeCommunicator.js

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { xbOrigin } from "./configs";
+
+export default class BridgeCommunicator extends React.PureComponent {
+  static propTypes = {
+    campaignReference: PropTypes.string,
+    onLoginStatusChange: PropTypes.func.isRequired,
+    onCookieOptionsLoad: PropTypes.func.isRequired,
+    handleBridgeError: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    campaignReference: null,
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Register with XcooBee domain in iframe
+    window.addEventListener("message", this.XcooBeeHandleResponse, false);
+  }
+
+  // This function is called when we have a response from XcooBee Domain
+  XcooBeeHandleResponse = (event) => {
+    const { onLoginStatusChange, onCookieOptionsLoad, handleBridgeError } = this.props;
+
+    if (event.origin === xbOrigin) {
+      const data = JSON.parse(event.data);
+      const action = Object.keys(data)[0];
+
+      if (action === "loginstatus") {
+        onLoginStatusChange(data[action]);
+
+        if (data[action]) {
+          this.fetchCookieOptions();
+        }
+      } else if (action === "cookieoptions") {
+        onCookieOptionsLoad(data[action]);
+      } else if (action === "savecookieconsent") {
+        // console.log("Cookie consents have been successfully saved.");
+      } else {
+        handleBridgeError(data[action]);
+      }
+    }
+  };
+
+  // Make a call to iframe
+  checkLoginStatus = () => {
+    const { campaignReference } = this.props;
+    const myMsgObj = {
+      action: "loginstatus",
+      campaign: campaignReference,
+      domain: window.location.origin,
+    };
+
+    this.frameRef.contentWindow.postMessage(JSON.stringify(myMsgObj), xbOrigin);
+  };
+
+  fetchCookieOptions() {
+    const { campaignReference } = this.props;
+    const myMsgObj = {
+      action: "cookieoptions",
+      action_params: {
+        type: "all",
+      },
+      campaign: campaignReference,
+      domain: window.location.origin,
+    };
+
+    this.frameRef.contentWindow.postMessage(JSON.stringify(myMsgObj), xbOrigin);
+  }
+
+  saveCookieConsents(cookieConsents) {
+    const { campaignReference } = this.props;
+    const myMsgObj = {
+      action: "savecookieconsent",
+      action_params: {
+        categories: cookieConsents,
+      },
+      campaign: campaignReference,
+      domain: window.location.origin,
+    };
+
+    this.frameRef.contentWindow.postMessage(JSON.stringify(myMsgObj), xbOrigin);
+  }
+
+  render() {
+    return (
+      <iframe
+        title="cookieBridge"
+        ref={(frameRef) => {
+          this.frameRef = frameRef;
+        }}
+        src="https://testapp.xcoobee.net/scripts/cookie-bridge/index.html"
```

Comment:
I assume you have a plan to change this to the production URL in the future? Or should this be using `xbOrigin`?

Reply:
Good catch, thanks.